File: spine-flutter/example/lib/simple_animation.dart
import 'package:spine_flutter/spine_flutter.dart';
import 'package:flutter/material.dart';

class SimpleAnimation extends StatelessWidget {
  const SimpleAnimation({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    reportLeaks();
    final controller = SpineWidgetController(onInitialized: (controller) {
      // Set the walk animation on track 0, let it loop
      controller.animationState.setAnimationByName(0, "walk", true);
    });

    return Scaffold(
      appBar: AppBar(title: const Text('Simple Animation')),
      body: SpineWidget.fromAsset("assets/spineboy.atlas", "assets/spineboy-pro.skel", controller)
    );
  }
}

File: spine-ios/Example/Spine iOS Example/SimpleAnimation.swift
import SwiftUI
import Spine

struct SimpleAnimation: View {
    
    @StateObject
    var controller = SpineController(
        onInitialized: { controller in
            controller.animationState.setAnimationByName(
                trackIndex: 0,
                animationName: "walk",
                loop: true
            )
        }
    )
    
    var body: some View {
        SpineView(
            from: .bundle(atlasFileName: "spineboy.atlas", skeletonFileName: "spineboy-pro.skel"),
//            from: .http(
//                atlasURL: URL(string: "https://github.com/denrase/spine-runtimes/raw/spine-ios/spine-ios/Example/Spine%20iOS%20Example/Assets/spineboy/spineboy.atlas")!,
//                skeletonURL:  URL(string: "https://github.com/denrase/spine-runtimes/raw/spine-ios/spine-ios/Example/Spine%20iOS%20Example/Assets/spineboy/spineboy-pro.skel")!
//            ),
            controller: controller,
            mode: .fit,
            alignment: .center
        )
        .navigationTitle("Simple Animation")
        .navigationBarTitleDisplayMode(.inline)
    }
}

#Preview {
    SimpleAnimation()
}
